Memory_Management/Paging.md

```diff
@@ -13,13 +74,91 @@ To implement paging, is strongly reccomended to have implemented exceptions too.
+* Page Table (PT) Every entry in the page Table points to a 4k memory page.
+
+Is important to note that the x86_64 architecture support mixing page size.
+
+Let's see first the structure of all entries type, and then explain the bits
+
+### PML4 & PDPR & PD
+
+PML4 and PDPR entry structure are identical, while the PD one has few differences. Let's start seeing the structure of the first two types: 
+
+|63     | 62        | 51                   | 39                     | 11  ...  9 |
+|-------|-----------|----------------------|------------------------|------------|
+|**EXB**| Available | _Reserved must be 0_ | **Table Base Address** | Available  |
+
+
+|8   ...   6 | 5     |  4      |  3      |  2      |  1      | 0     |
+|------------|-------|---------|---------|---------|---------|-------|
+| _Reserved_ | **A** | **PCD** | **PWT** | **U/S** | **R/W** | **P** |
+
+Where **Table Base Address** is PDPR Table base address if the table is PML4 or the PD base address if the table is the PDPR.
+
+Now the Page directory has few differences: 
+* Bits 39 to 12 are the Page Table base address when using 4k pages, or 2 MB Memory area if the PS bit is set.
+* Bits 6 and 8 must be 0
+* Bit 7 (the PS) must be 1
+* If we are using 2MB Pages bit 12 to 20 are reserved and must be 0. If not it will cause a #PF
+
+### Page Table 
+A page table entry structure is still similar to the one above, but it contains few more bits that can be set: 
+
+|63     | 62    | 51         | 39                    | 11  ... 9 |
+|-------|-------|------------|-----------------------|-----------|
+|**EXB**| Avail | _Reserved must be 0_ | **Page Base Address** | Available |
+
+
+| 8     | 7       | 6      | 5     |  4      |  3      |  2      |  1      | 0     |
+|-------|---------|--------|-------|---------|---------|---------|---------|-------|
+| **G** | **PAT** | **D**  | **A** | **PCD** | **PWT** | **U/S** | **R/W** | **P** |
+
+
+In this table there are 3 new bits (D, PAT, G) and the Page Base Address as already explained is not pointing to a table but to a memory area. 
+
+In the next section we will se all the fields of an entry.
+
+### Page Table entries fields
+
+Below is a list of all the fields present in the table entries, with an explanation of the most commonly used.
+
+* **P** (Present): If set this tells the CPU that the current page or page table pointed by this entry is currently loaded in physical memory, so when accessed the address translation can be carried out. If is 0 this means that the page is not loaded into memory so a virtual address that contains this entry will cause a Page Fault.
+* **R/W** (Read/Write): If set the page can be both read and written, if clear is in read only mode. When this bit is set in a page Directory it tells the cpu that all its entries will share that setting
+* **User/Supervisor** It describe the privilege level, if clear the page has the Supervisor level, while if it is set the level is Supervisor.
```

Comment:
haha I think you meant if it is clear, it's a user level page. Worth mentioning that supervisor pages can only be accessed by CPL > 3 (rings 0/1/2), and user levels can be accessed by any CPL. aka setting this bit means ring 3 code and data can access this page.